orderer/consensus/smartbft/chain.go

```diff
@@ -419,6 +427,72 @@ func (c *BFTChain) HandleMessage(sender uint64, m *smartbftprotos.Message, metad
 	c.consensus.HandleMessage(sender, m)
 }
 
+func (c *BFTChain) getSuspectsFromSignatures(signatures []*smartbftprotos.Signature) []int32 {
+	var allSuspects []int32
+	for _, signature := range signatures {
+		sig := &Signature{}
+		if err := sig.Unmarshal(signature.Msg); err != nil {
+			c.Logger.Warningf("Failed unmarshaling signature")
+			continue
+		}
+		aux := sig.CommitteeAuxiliaryInput
+		committeeFeedback := &smartbft2.CommitteeFeedback{}
+		if err := proto.Unmarshal(aux, committeeFeedback); err != nil {
+			c.Logger.Warningf("Failed unmarshaling committeeFeedback")
+			continue
+		}
+		list := committeeFeedback.Suspects
+		cleanList := removeDuplicates(list)
+		allSuspects = append(allSuspects, cleanList...)
+	}
+	return agreedSuspects(allSuspects, 1) // TODO use f
+}
+
+func (c *BFTChain) getSuspectsFromBlock(proposal *smartbftprotos.Proposal) (bool, []int32) {
+	tuple := &ByteBufferTuple{}
+	if err := tuple.FromBytes(proposal.Payload); err != nil {
+		c.Logger.Warningf("Failed reading proposal payload")
+		return false, nil
+	}
+	metadata := &common.BlockMetadata{}
+	if err := proto.Unmarshal(tuple.B, metadata); err != nil {
+		c.Logger.Warningf("Failed unmarshaling block metadata")
+		return false, nil
+	}
+	signatureMetadata := &common.Metadata{}
```

Comment:
looks like we are missing a range check for common.BlockMetadataIndex_SIGNATURES?